inc/class_datacache.php

```diff
@@ -450,6 +462,113 @@ function size_of($name='')
 		}
 	}
 
+	/**
+	 * Test if the cache store can work properly.
+	 *
+	 * @return bool True on yes, false on no.
+	 */
+	function test_availability()
+	{
+		$content = md5((string) constant("TIME_NOW"));
+
+		if($this->handler instanceof CacheHandlerInterface)
+		{
+			$this->handler->put($content, $content);
+			$content_stored = $this->handler->fetch($content);
+			$this->handler->delete($content);
+			return $content_stored == $content;
+		}
+
+		// Always return false if no external cache store is used.
+		return false;
+	}
+
+	/**
+	 * Check cache from files or db, and recreate it if it's missing.
+	 *
+	 * @param string $package The cache type, 'core', 'core_*', 'plugin_*', 'all'.
+	 * @param array $name The cache components to check.
+	 * @return bool True on successfully rebuilt cache, false otherwise.
+	 */
+	function check_cache($package = 'core', $cache_names = array())
+	{
+		// Ensure the value is fetched from the original datacache table.
+		$this->read('cache_checked', true);
+
+		// Only continue check if forced and cache is out-dated or non-existent.
+		if(isset($this->cache['cache_checked']) && TIME_NOW - $this->cache['cache_checked'] < 60 * 10)	//TODO: hard coded time?
```

Comment:
I think a hard-coded time is fine, though perhaps define it as a constant rather than magic numbers?